daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -824,10 +826,10 @@ private[lf] object Speedy {
 
   object Machine {
 
-    private val damlTraceLog = LoggerFactory.getLogger("daml.tracelog")
-    private val damlWarnings = LoggerFactory.getLogger("daml.warnings")
+    private val damlTraceLog = ContextualizedLogger.damlTraceLog
+    private val damlWarnings = ContextualizedLogger.damlWarnings
```

Comment:
Yeah, it looks like it's more straightforward to use `ContextualizedLogger` just as you were using `LoggerFactory`, without pushing specific names to the underlying library.

Reply:
`ContextualizedLogger.createFor` is not public. 
It is maybe fine to use `ContextualizedLogger.createFor(getClass)` for the warnings, but I do not think it is fine for `damlTraceLog`

Reply:
Thanks for mentioning this, I didn't realize. I think that the `createFor` that takes a string can be safely made public.

Reply:
On the other hand, the one that takes a logger should be kept package private and reserved for testing.

Reply:
:+1:

Reply:
One question. I see that there is a cache for the call to `get`. 
I think if I make `createFor` public,  should I move the cache to this method ?